lib/team_statistics.rb

```diff
@@ -25,163 +25,100 @@ def games_played(team_id)
     end
   end
 
-  def best_season(team_id)
-    # Season with the highest win percentage for a team.
-    # Find all home and away games for team_id
-    # Group by season
-    # Compare wins vs total games for each season
-    # Find max win percentage
+  def season_best_and_worst(team_id, best_or_worst)
     games_played_in = games_played(team_id)
     games_by_season = games_played_in.group_by {|game| game.season}
+
     win_percentage_by_season = games_by_season.transform_values do |season|
-      games_won = 0
-
-      season.each do |game|
-        if team_id == game.away_team_id && game.away_goals > game.home_goals
-          games_won += 1
-        elsif team_id == game.home_team_id && game.home_goals > game.away_goals
-          games_won += 1
-        end
-      end
+      games_won = count_wins(team_id, season)
       (games_won/season.length.to_f).round(2)
     end
-    best_season = win_percentage_by_season.max_by { |season| season[1]}
-    best_season[0]
-  end
 
-  def worst_season(team_id)
-    # Season with the lowest win percentage for a team.
-    # Find all home and away games for team_id
-    # Group by season
-    # Compare wins vs total games for each season
-    # Find min win percentage
-
-    games_played_in = games_played(team_id)
-    games_by_season = games_played_in.group_by {|game| game.season}
-    win_percentage_by_season = games_by_season.transform_values do |season|
-      games_won = 0
-
-      season.each do |game|
-        if team_id == game.away_team_id && game.away_goals > game.home_goals
-          games_won += 1
-        elsif team_id == game.home_team_id && game.home_goals > game.away_goals
-          games_won += 1
-        end
-      end
-      (games_won/season.length.to_f).round(2)
+    if best_or_worst == "best"
+      best_season = win_percentage_by_season.max_by { |season| season[1]}
+      best_season[0]
```

Comment:
Might be able to put this in parent class too, I have a helper method that does something similar.